models/shopify__discounts.sql

```diff
@@ -0,0 +1,90 @@
+with discount as (
+
+    select *
+    from {{ var('shopify_discount_code') }}
+),
+
+price_rule as (
+
+    select *
+    from {{ var('shopify_price_rule') }}
+),
+
+orders_aggregated as (
+
+    select *
+    from {{ ref('int_shopify__discounts__order_aggregates')}}
+),
+
+abandoned_checkouts_aggregated as (
+
+    select *
+    from {{ ref('int_shopify__discounts__abandoned_checkouts')}}
+),
+
+discount_price_rule_joined as (
+
+    select
+        discount.*,
+        price_rule.target_selection,
+        price_rule.target_type,
+        price_rule.title,
+        price_rule.usage_limit,
+        price_rule.value,
+        price_rule.value_type,
+        price_rule.allocation_limit,
+        price_rule.allocation_method,
+        price_rule.is_once_per_customer,
+        price_rule.customer_selection,
+        -- the below are NULL if customer_selection = all
+        price_rule.prereq_min_quantity,
+        price_rule.prereq_max_shipping_price,
+        price_rule.prereq_min_subtotal,
+        price_rule.prereq_min_purchase_quantity_for_entitlement,
```

Comment:
It looks like this can also have custom field variations? Is this something we should add a variable for to passthrough if necessary? What are your thoughts?
![image](https://user-images.githubusercontent.com/74217849/211907109-ee2fa82b-fc7f-4fba-957d-b66c04e8dd44.png)

Reply:
ah good catch. if the custom fields are only related to prerequisite entitlement, i don't see them being _critical_, but i don't see a downside to including support for the passthrough columns  🤷‍♂️

Reply:
Yeah I was considering how critical they would be if we added them. It is also hard to understand how people are using them without the customer context. Let's exclude the passthrough for now, but can see how users engage. If it is requested we can explore adding the flexibility into the model.